auth/access.go

```diff
@@ -17,13 +40,121 @@ func (a AccessToken) Value() string {
 	return a.value
 }
 
-func NewAccessToken(token string) system.Credential {
-	if token != "" {
-		return &AccessToken{
-			scheme: "Bearer",
-			value:  token,
-		}
+type (
+	AuthorizationRequest struct {
+		Action   uint8    `json:"action"`
+		Resource []string `json:"resource"`
 	}
 
-	return nil
+	AuthorizationResponse struct {
+		Identity    string `json:"identity"`
+		Account     string `json:"account"`
+		Environment string `json:"environment"`
+		Tenant      string `json:"tenant"`
+		Path        string `json:"path"`
+	}
+)
+
+func NewAuthorizationContext(response *AuthorizationResponse) *AuthorizationContext {
+	segments := strings.Split(response.Path, "/")
+	return &AuthorizationContext{
+		account:     response.Account,
+		environment: response.Environment,
+		tenant:      response.Tenant,
+		namespace:   segments[0],
+		module:      segments[1],
+	}
+}
+
+type AuthorizationContext struct {
+	account     string
+	environment string
+	tenant      string
+	namespace   string
+	module      string
+}
+
+func (authz *AuthorizationContext) Identity() string {
+	return authz.account
+}
+
+func (authz *AuthorizationContext) Account() string {
+	return authz.account
+}
+
+func (authz *AuthorizationContext) Environment() string {
+	return authz.environment
+}
+
+func (authz *AuthorizationContext) Tenant() string {
+	return authz.tenant
+}
+
+func (authz *AuthorizationContext) Namespace() string {
+	return authz.namespace
+}
+
+func (authz *AuthorizationContext) Module() string {
+	return authz.module
+}
+
+const (
+	AuthorizationCtxKey = "authorization"
+	AccessExecute       = 8
+)
+
+func AuthorizationMiddleware(client *http.Client, controlplane string) vk.Middleware {
+	authority := fmt.Sprintf("%s/%s", controlplane, "api/v2/access")
+	return func(req *http.Request, ctx *vk.Ctx) error {
+		identifier := ctx.Params.ByName("ident")
```

Comment:
So this won't work on `/ref/:ref` calls (for now)?

Reply:
I believe we decided that this was okay on Twist.